### PR TITLE
1407-besso-migrate-to-knex-post-dashboard-auth-login-endpoint

### PR DESCRIPTION
**Pull request by paula-uxengineer**
### This pull request includes the following changes:

**1 - Migration to Knex:**
The POST /dashboard/auth/login endpoint has been refactored to use Knex for database operations. This migration ensures a more consistent and efficient way to handle SQL queries across the application.

**2 - Test Configuration for Knex:**
Added the necessary configuration for Knex to support the testing environment. This includes the setup for the test database connection, ensuring that all tests can be executed against a properly configured database instance.

### CODE REVIEW

File: services/sso/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,12 @@ All notable changes to this project will be documented in this file.
 
 # Changelog
 
+## [1.34.2] - 2024-06-19
```
Comment (joanbr4) [inline services/sso/CHANGELOG.md:5]: Pls update to this time, not far far away

**Review by joanbr4 — APPROVED** [review]

File: services/sso/src/controllers/dashboard/auth/login.ts [hunk 2]
```diff
@@ -12,14 +12,12 @@ import { appConfig } from '../../../config'
...
-  const user = userResult.rows[0] as Pick<
-    User,
-    'id' | 'password' | 'status' | 'role'
-  >
+  const userResult = await db('user')
+    .select('id', 'password', 'status', 'role')
+    .where('dni', dniToUpperCase)
+    .first()
+
+  const user = userResult as Pick<User, 'id' | 'password' | 'status' | 'role'>
```
Comment (xpan1c) [inline services/sso/src/controllers/dashboard/auth/login.ts:20]:
```suggestion
  const user = await db('user')<User>
    .select('id', 'password', 'status', 'role')
    .where('dni', dniToUpperCase)
    .first()
```
You can infer the type this way as well, [Knex.js documentation](https://knexjs.org/guide/query-builder.html#usage-with-typescript-1)

**Review by xpan1c — APPROVED** [review]

**Review by kevinmamaqi — DISMISSED** [review]
hacer cambios

**Comment by paula-uxengineer** [comment]
> hacer cambios

he hecho los cambios

**Review by kevinmamaqi — APPROVED** [review]

**Review by xpan1c — APPROVED** [review]

**Pull request merged**